src/mlir/ast-utils.h

```diff
@@ -151,7 +154,13 @@ namespace mlir::verona
     /// Return true if node is a value
     static bool isConstant(::ast::WeakAst ast)
     {
-      return isValue(ast) && isAny(ast, {NodeKind::Integer, NodeKind::Float});
+      return isValue(ast) &&
```

Comment:
Shouldn't string literals be included? Or are those not fully implemented yet?

Reply:
Not implemented yet. Strings in MLIR are a major PITA, I'll leave it for when we really need it...

Reply:
Strings are also complex to do right in the language. We want to avoid baking the standard library string type into the language if at all possible. I would like to have a sequence-of-UTF8-code-units literal that can be converted to string types, in the same way that we have a number literal that can be converted to any kind of integer, but I’m not sure how feasible that is. I’d also like to have a string interface in the standard library and avoid taking a dependency on a concrete string type anywhere. For low-level code, we will want a string that is a simple array of 8-bit values, for higher-level code we will want a rich Unicode string representation. The C++ standard library bakes a load of locale stuff in, which makes it very hard to run in embedded environments (e.g. in a kernel) where you expect to generate things that another component will localise if necessary. C goes in the opposite direction and gives a string that is unusable in any higher level applications (and is hard to optimise). Rust has many strings but doesn’t make it easy to mix them.